ang/civicase-features/quotations/directives/quotations-view.directive.js

```diff
@@ -0,0 +1,65 @@
+(function (angular, _) {
+  var module = angular.module('civicase-features');
+
+  module.directive('quotationsView', function () {
+    return {
+      restrict: 'E',
+      controller: 'quotationsViewController',
+      templateUrl: '~/civicase-features/quotations/directives/quotations-view.directive.html',
+      scope: {
+        salesOrderId: '='
+      }
+    };
+  });
+
+  module.controller('quotationsViewController', quotationsViewController);
+
+  /**
+   * @param {object} crmApi4 api V4 service
+   * @param {object} $scope the controller scope
+   * @param {object} CaseUtils case utility service
+   * @param {object} CurrencyCodes CurrencyCodes service
+   */
+  function quotationsViewController (crmApi4, $scope, CaseUtils, CurrencyCodes) {
+    $scope.taxRates = [];
+    $scope.currencySymbol = '';
+    $scope.dashboardLink = '#';
+    $scope.salesOrder = {
+      total_before_tax: 0,
+      total_after_tax: 0
+    };
+    $scope.hasCase = false;
+
+    (function init () {
+      if ($scope.salesOrderId) {
+        getSalesOrderAndLineItems();
+      }
+    })();
+
+    /**
+     * Retrieves the sales order and its line items from API
+     */
+    function getSalesOrderAndLineItems () {
+      crmApi4('CaseSalesOrder', 'get', {
+        select: ['*', 'case_sales_order_line.*', 'client_id.display_name', 'owner_id.display_name', 'case_id.case_type_id:label', 'case_id.subject', 'status_id:label'],
+        where: [['id', '=', $scope.salesOrderId]],
+        limit: 25,
```

Comment:
Don't see the reason to limit the result to 25, it should be `0` instead so all the results are retrieved

Reply:
thanks chief, I'm actually expecting 1 result at most, since I'm filtering by ID, so I've updated the limit to 1.